packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -982,6 +982,14 @@ class _NestedScrollCoordinator implements ScrollActivityDelegate, ScrollHoldCont
         maxRange = pixels + extra;
         assert(minRange <= maxRange);
         correctionOffset = _outerPosition!.pixels - pixels;
+
+        // Compute IEEE-754 error that might potentially occur and cause position
+        // to overflow when running the simulation.
+        final double ieee754Error = maxRange + correctionOffset - _outerPosition!.maxScrollExtent;
+        if (ieee754Error != 0.0) {
```

Comment:
We usually check for the error this way:
```
if ((a-b).abs >precisionErrorTolerance)
```
and we should not try to correct the position error, because sometime it can't be corrected. For example you may see that
```
maxRange - ieee754Error == maxRange
```
returns true.

we should make the function to still work even if the number is off by a bit.

Reply:
So this ends up in ClampingScrollPhysics which should allow this precision tolerance
I'm wondering whether `outOfRange` and `atEdge` should do that?
Also are there any other places you can imagine this should be respected?

https://github.com/flutter/flutter/blob/d637b1874f7d66405cc1ebe33d48acfa1e80a318/packages/flutter/lib/src/widgets/scroll_metrics.dart#L96-L102

Reply:
It is hard to capture every instance this can happen. I have personally fixed a bunch of places. If the issue happens around this two getter, we can do something like

```dart
bool get outOfRange => minScrollExtent - pixels > -precisionErrorTolerance || pixels - maxScrollExtent >  -precisionErrorTolerance; 
```